.github/workflows/check.yaml

```diff
@@ -0,0 +1,39 @@
+name: Check workflow running linter, unit and functional tests
+
+on:
+  workflow_call:
+  pull_request:
+    types: [ opened, synchronize, reopened ]
+    branches: [ master, main ]
+    paths-ignore:
+      - '**.md'
+      - '**.rst'
+
+jobs:
+  lint-unit:
+    uses: canonical/bootstack-actions/.github/workflows/lint-unit.yaml@v2
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ['3.8', '3.10']
+    with:
+      python-version: ${{ matrix.python-version }}
+      tox-version: '<4'
+
+  func:
+    uses: canonical/bootstack-actions/.github/workflows/func.yaml@v2
+    needs: lint-unit
+    strategy:
+      fail-fast: false
+      matrix:
+        series: ["focal", "jammy"]
+        include:
+          - juju-channel: '3.1/stable'
+            command: 'make functional'
```

Comment:
These doesn't need to be put in the `include` field since they are the same for both series. Simply put their values under `with` is enough (as suggested below).

Reply:
As I understand, v2 is for juju3.1. Could we change the default value of juju-channel to 3.1? Then we don't change the channel one - by - one in the future.. e.g. 3.2, 3.3 ..
https://github.com/canonical/bootstack-actions/blob/v2/.github/workflows/func.yaml

Reply:
I'm +1 on changing the default of `juju-channel` to `3.1/stable` for the v2 cahnnel.

Reply:
I opened an issue in bootstack-actions for this: https://github.com/canonical/bootstack-actions/issues/34. IMO we can merge this PR regardless since this discussion is out of the current scope.